addons/website/models/website_page_properties.py

```diff
@@ -168,6 +169,18 @@ def _compute_is_homepage(self):
             current_homepage_url = record.website_id.homepage_url or '/'
             record.is_homepage = url == current_homepage_url
 
+    @api.onchange('has_parent_page')
+    def _onchange_has_parent_page(self):
+        website = self.env['website'].get_current_website()
+        homepage = self.env['website.page'].search([('website_id', '=', website.id)]).filtered(lambda r: r.is_homepage)
+        for page in self:
+            page.parent_page_id = page.parent_page_id or homepage if page.has_parent_page else False
```

Comment:
Add parenthesis, reading this I actually have no clue if you meant `(a or b) if condition else c` or `a or (b if condition else c)`... I learned that it is the first one that applied.

Also it would not hurt to have some comments to explain what this is about.